nat-lab/tests/mesh_api.py

```diff
@@ -353,12 +354,54 @@ def generate_peer_config(node: Node, allowed_ips: str) -> str:
                 node, ", ".join(cls.get_allowed_ip_list(node.ip_addresses))
             )
 
-        full_command = (
-            f"docker exec -d --privileged {server_config['container']} bash -c 'echo"
-            f' "{wg_conf}" > /etc/wireguard/wg0.conf; wg-quick down'
-            " /etc/wireguard/wg0.conf; wg-quick up /etc/wireguard/wg0.conf'"
-        )
-        os.system(full_command)
+            def exec_docker_commands(
+                container_id: str, commands: typing.List[str]
+            ) -> None:
+                for command in commands:
+                    cmd = f"docker exec -d --privileged {container_id} bash -c '{command}'"
+                    os.system(cmd)
+
+            if "type" in server_config and server_config["type"] == "nordlynx":
+                priv_key = server_config["private_key"]
+
+                # TODO: would be nice to move link delete to some teardown step
+                nlx_commands = [
+                    "mkdir -p /etc/nordlynx",
+                    f'echo "{priv_key}" > /etc/nordlynx/private.key',
+                    "pkill nlx-radius",
+                    "pkill fakefm",
+                    # Order of operations for Nordlynx is important
+                    "ip link delete nordlynx0",
+                    "ip link add dev nordlynx0 type nordlynx",
+                    "ip addr add 10.5.0.1/16 dev nordlynx0",  # VPN
+                    "ip addr add 100.64.0.1/10 dev nordlynx0",  # meshnet
+                    "ip link set mtu 1420 dev nordlynx0",
+                    "ip link set dev nordlynx0 up",
+                    "nlx set nordlynx0 listen-port 51820",
+                    "nlx set nordlynx0 private-key /etc/nordlynx/private.key",
+                    # setup proper routing on the VPN(nordlynx) server
+                    (
+                        "iptables -A FORWARD -i nordlynx0 -j ACCEPT; iptables -A"
+                        " FORWARD -o nordlynx0 -j ACCEPT; iptables -t nat -A"
+                        " POSTROUTING -o eth0 -j MASQUERADE"
+                    ),
+                    # courtesy sleep so it would be really all up and running
+                    "sleep 2",
+                    "ip addr s > /var/log/ipaddr.show",
+                    "nlx > /var/log/nlx.log",
+                    "/usr/bin/fakefm/fakefm -s testing123 2>&1 > /var/log/fakefm.log",
+                    "/opt/nlx-radius/nlx-radius --log=stdout --verb=DEBUG 2>&1 > /var/log/nordlynx.log",
```

Comment:
I would consider moving this commands to some script

Reply:
:100: agree with @stalowyjez

Reply:
So it's tricky. I've tried doing that however nordlynx is finicky for order of setup so it would need to be like 2 separate parts in the separate shell script and some steps inside of the python. While it would make it a bit simpler I think debugging it would become harder. I could at the very least move it to a separate function so not to pollute this one as much but then this whole `setup_vpn_servers` function would become just a simple wrapper around it

Reply:
Hmm, I don't get this tbh, here you have just a list of commands which you run one by one with `docker exec` - why couldn't you run just `docker exec -d --privileged server_container /opt/bin/some_script.sh` instead?

Reply:
Actually it might be as easy as you describe. I was a bit misguided, let me try :D

Reply:
@stalowyjez @gytsto so it worked pretty nicely. I moved the whole script to a separate shellscript referred in docker-compose and now python code only has private key output to a file and `nlx set ... private-key` command